bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/SAICiSMARTBridgeHandler.java

```diff
@@ -0,0 +1,296 @@
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.GsonBuilder;
+
+import net.heberling.ismart.asn1.v1_1.Message;
+import net.heberling.ismart.asn1.v1_1.MessageCoder;
+import net.heberling.ismart.asn1.v1_1.entity.AlarmSwitch;
+import net.heberling.ismart.asn1.v1_1.entity.AlarmSwitchReq;
+import net.heberling.ismart.asn1.v1_1.entity.MP_AlarmSettingType;
+import net.heberling.ismart.asn1.v1_1.entity.MP_UserLoggingInReq;
+import net.heberling.ismart.asn1.v1_1.entity.MP_UserLoggingInResp;
+import net.heberling.ismart.asn1.v1_1.entity.MessageListReq;
+import net.heberling.ismart.asn1.v1_1.entity.MessageListResp;
+import net.heberling.ismart.asn1.v1_1.entity.StartEndNumber;
+import net.heberling.ismart.asn1.v1_1.entity.VinInfo;
+
+/**
+ * The {@link SAICiSMARTBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Heberling - Initial contribution
+ */
+@NonNullByDefault
+public class SAICiSMARTBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SAICiSMARTBridgeHandler.class);
+
+    private @Nullable SAICiSMARTBridgeConfiguration config;
+
+    private @Nullable String uid;
+
+    private @Nullable String token;
+
+    private @Nullable Collection<VinInfo> vinList;
+    private HttpClient httpClient;
+    private @Nullable Future<?> pollingJob;
+
+    public SAICiSMARTBridgeHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands available
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SAICiSMARTBridgeConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Validate configuration
+        if (this.config.username.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "iSMART Username is empty!");
+            return;
+        }
+        if (this.config.password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "iSMART Password is empty!");
+            return;
+        }
+        if (this.config.username.length() > 50) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "iSMART Username too long!");
+            return;
+        }
+        pollingJob = scheduler.scheduleWithFixedDelay(this::updateStatus, 1,
+                SAICiSMARTBindingConstants.REFRESH_INTERVAL, TimeUnit.SECONDS);
+    }
+
+    private void updateStatus() {
+        if (uid == null || token == null) {
```

Comment:
Maybe i missed it, but what about an expiring token? Can;t see some check to see if it is allmost expired and renewed.

Reply:
Thi sis not yet adressed

Reply:
I am not aware that the token expires.  This has not happened in my testing.

There is a expire timestamp in the login response.
https://github.com/SAIC-iSmart-API/documentation/blob/137af4b4fa3580f2cb3d98d61e921954110599ad/ASN.1%20schema/v1_1/ApplicationData.asn1#L39

Maybe @tisoft can advise if and how this should be handled?

Reply:
Let's assume the token does not expire for others too. If there is a need for a check and fix it can probably be implemented easily in a future PR.